prov/opx/include/rdma/opx/fi_opx_hfi1_packet.h

```diff
@@ -43,7 +43,11 @@
 #include "rdma/fabric.h"	/* only for 'fi_addr_t' ... which is a typedef to uint64_t */
 #include "rdma/opx/fi_opx_addr.h"
 
-
+#if defined(__riscv) && defined(__riscv_xlen) && (__riscv_xlen == 64)
+#ifndef PAGE_SIZE
+/* 4K pages default */
+#define PAGE_SIZE 4096
```

Comment:
In light of fixing up the config.m4, is defining PAGE_SIZE still necessary for riscv?

Reply:
@belynam I'm not able to get that working on the autoconf side of things. Is this something ya'll could fix on your side after a PR merge?

Reply:
We're picking up `PAGE_SIZE` as defined in **/usr/include/sys/user.h** via `#include <sys/user.h>` in fi_opx_hfi1_packet.h. Perhaps the system you're building on doesn't have it defined in there?

In any case, I think it's fine to leave this as is if needed for riscv. I don't consider this a blocker to merging.

Reply:
Ah, that explains it. I'm on a riscv64 ubuntu distribution and `/usr/include/sys` does not exist.

Reply:
@belynam I could modify the source to use `getpagesize` not sure what the pros/cons are to that approach.

Reply:
> I could modify the source to use getpagesize not sure what the pros/cons are to that approach.

I think for now, just leave it as you have it in this PR (defining PAGE_SIZE as 4K for riscv). It looks to me like we shouldn't be getting PAGE_SIZE from /usr/include/sys/user.h. We should be using `sysconf(_SC_PAGESIZE)` instead. But we'll take care of fixing that, as there are bound to be performance impacts.

Reply:
thanks! PR has been updated with the latest changes!